src/MagicWormhole/Internal/FileTransfer.hs

```diff
@@ -29,12 +29,26 @@ import System.Posix.Types (FileOffset)
 data Offer
   -- | A simple text message.
   = Message Text
+  -- | Offer a File with filename and size.
   | File FilePath FileOffset
+  -- | Offer a Directory
+  | Directory
+    Text
+    -- ^ Mode. Currently always "zipfile/deflated".
+    Text
+    -- ^ Directory Name.
+    Integer
+    -- ^ size of the transmitted compressed data in bytes
+    Integer
+    -- ^ estimated total size of the uncompressed directory
+    Integer
+    -- ^ number of files and directories being sent
```

Comment:
Personally, I think this is getting big enough that the fields should be named.

Reply:
FWIW, I meant named fields (e.g. `numFiles :: Int64`), not named types. I'm not convinced the new type aliases add much value.